src/main/java/org/folio/rs/controller/PubSubEventController.java

```diff
@@ -0,0 +1,33 @@
+package org.folio.rs.controller;
+
+import static org.folio.rs.util.MapperUtils.mapJsonToMovedEventRequest;
+
+import java.util.Objects;
+import javax.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.rs.domain.dto.MovedEventRequest;
+import org.folio.rs.rest.resource.PubSubHandlersApi;
+import org.folio.rs.service.RetrievalQueueService;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+@Log4j2
+@Controller
+@RequiredArgsConstructor
+@RequestMapping(value = "/remote-storage/")
+public class PubSubEventController implements PubSubHandlersApi {
+
+  private RetrievalQueueService retrievalQueueService;
```

Comment:
Missing `final` keyword, `@RequiredArgsConstructor` annotation will work only for final fields and fields with constraints such as `@NonNull`.